core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -836,9 +830,10 @@ private[spark] class ExecutorAllocationManager(
 
     /**
      * The number of tasks currently running across all stages.
+     * Include running-but-zombie stage attempts
      */
     def totalRunningTasks(): Int = {
-      stageIdToNumRunningTask.values.sum
+      liveTaskIds.size
```

Comment:
If you only want to count the total number of active tasks why not count from `executorIdToTaskIds` ? This way you don't need to store any other data.